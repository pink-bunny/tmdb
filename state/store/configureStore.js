import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
