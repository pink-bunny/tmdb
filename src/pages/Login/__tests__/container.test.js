import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedLogin, { handleSubmit } from '../container';

describe('Login container', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();
  const wrapper = shallow(<ConnectedLogin store={store} />);

  it('matches snapshot', () => {
    const container = wrapper.dive().dive().dive();

    expect(container).toMatchSnapshot();
  });

  describe('handleSubmit()', () => {
    it('calls requestSession with arguments', () => {
      const props = { requestSession: jest.fn() };
      const setErrors = jest.fn();
      const setSubmitting = jest.fn();
      const values = {};
      handleSubmit(values, { props, setErrors, setSubmitting });

      expect(props.requestSession).toHaveBeenCalledWith(values, setErrors, setSubmitting);
    });
  });

  describe('validationSchema', () => {
    const container = wrapper.dive().props().validationSchema().describe();

    it('matches snapshot', () => {
      expect(container).toMatchSnapshot();
    });
  });
});
