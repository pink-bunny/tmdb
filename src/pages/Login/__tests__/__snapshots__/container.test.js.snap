// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "fields": Object {
    "password": Object {
      "label": undefined,
      "meta": undefined,
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
        Object {
          "name": "min",
          "params": Object {
            "min": 4,
          },
        },
      ],
      "type": "string",
    },
    "username": Object {
      "label": undefined,
      "meta": undefined,
      "tests": Array [
        Object {
          "name": "required",
          "params": undefined,
        },
      ],
      "type": "string",
    },
  },
  "label": undefined,
  "meta": undefined,
  "tests": Array [],
  "type": "object",
}
`;

exports[`Login container matches snapshot 1`] = `
<Login
  dirty={false}
  errors={Object {}}
  getFieldHelpers={[Function]}
  getFieldMeta={[Function]}
  getFieldProps={[Function]}
  handleBlur={[Function]}
  handleChange={[Function]}
  handleReset={[Function]}
  handleSubmit={[Function]}
  initialErrors={Object {}}
  initialTouched={Object {}}
  initialValues={
    Object {
      "password": "",
      "username": "",
    }
  }
  isSubmitting={false}
  isValid={true}
  isValidating={false}
  registerField={[Function]}
  requestSession={[Function]}
  resetForm={[Function]}
  setErrors={[Function]}
  setFieldError={[Function]}
  setFieldTouched={[Function]}
  setFieldValue={[Function]}
  setFormikState={[Function]}
  setStatus={[Function]}
  setSubmitting={[Function]}
  setTouched={[Function]}
  setValues={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  submitCount={0}
  submitForm={[Function]}
  touched={Object {}}
  unregisterField={[Function]}
  validateField={[Function]}
  validateForm={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validateOnMount={false}
  values={
    Object {
      "password": "",
      "username": "",
    }
  }
/>
`;
